pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -702,6 +702,15 @@ def test_setitem_npmatrix_2d(self):
 
         tm.assert_frame_equal(df, expected)
 
+    @pytest.mark.parametrize("vals", [{}, {"d": "a"}])
+    def test_setitem_aligning_dict_with_index(self, vals):
+        # GH#47216
+        df = DataFrame({"a": [1, 2], "b": [3, 4], **vals})
+        df.loc[:, "a"] = {1: 100, 0: 200}
+        df.loc[:, "c"] = {0: 5, 1: 6}
```

Comment:
Can you add to the test where the dict is partial, e.g. `... = {0: 5}` (column e?)

Reply:
Done